### ISSUE TITLE
Komodo references system python on Linux (and possibly OSX)

### ISSUE THREAD

**Issue by Naatan** [Komodo/KomodoEdit#526]
On Komodo startup you see the following:

```
[2015-08-14 15:09:56,335] [INFO] Startup: /usr/bin/python built on Fri Jan 9 17:03:12 2015
```

`/usr/bin/python` is the [sys.executable value](https://github.com/Komodo/KomodoIDE/blob/master//src/components/koInfoService.p.py#L94).

This part in particular looks suspicious:

http://hg.mozilla.org/pyxpcom/file/55f2449179df/xpcom/src/dllmain.cpp#l167

Worth nothing - I run Arch Linux, `/usr/bin/python` is Python 3. There is absolutely no way Komodo would run if it's actually invoking python 3, but the reference is a reason for alarm and we should find out why it's happening.

**Comment by mitchell-as** [comment]
Seems to be a false positive. I've changed `self.buildASCTime` to `sys.version` here: https://github.com/Komodo/KomodoEdit/blob/8865f7249de2715c648b6aebc37b0f00b6466214/src/components/koInfoService.p.py#L118

and the version of Python Komodo uses (2.7.8) is different than my /usr/bin/python's version (2.7.6). Therefore Komodo is using its own Python. As for why this is happening is beyond me. Your hunch is quite reasonable. Our resident Python expert may have a better idea if you want to investigate further.

**Comment by Naatan** [comment]
Nah lets leave it for now, the issue that I thought was related tot his ended up being something completely different, so this is probably a non-issue. Thanks for checking :)

**Comment by Naatan** [comment]
Reopening as I've found confirmation that it's using the system python. The new docs addon uses sqlite3 with fts3 (python module). This is not part of our packaged Python but it works anyway on my linux environment, because Komodo is using my system python.

I opened a python console in Komodo and executed:

```
import sys
print sys.executable
/usr/bin/python 
```

Further adding to the proof that it's using the system python.

**Comment by mitchell-as** [comment]
Sorry, I still don't buy it. Here's some information on my System Python vs. Komodo's Python. Please do the same for your machine.

## System Pyshell

    >>> import sys
    >>> sys.executable
    '/usr/bin/python'
    >>> sys.version
    '2.7.8 (default, Nov 24 2014, 11:21:38) \n[GCC 4.8.2 20140120 (Red Hat 4.8.2-15)]'
    >>> sys.path
    ['', 
     '/usr/lib/python2.7',
     '/usr/lib/python2.7/plat-x86_64-linux-gnu',
     '/usr/lib/python2.7/lib-tk',
     '/usr/lib/python2.7/lib-old',
     '/usr/lib/python2.7/lib-dynload',
     '/usr/local/lib/python2.7/dist-packages',
     '/usr/lib/python2.7/dist-packages',
     '/usr/lib/python2.7/dist-packages/PILcompat',
     '/usr/lib/python2.7/dist-packages/gtk-2.0',
     '/usr/lib/python2.7/dist-packages/ubuntu-sso-client']

## Komodo Pyshell

    >>> import sys
    >>> sys.executable
    '/usr/bin/python'
    >>> sys.version
    '2.7.8 (default, Nov 24 2014, 11:21:38) \n[GCC 4.8.2 20140120 (Red Hat 4.8.2-15)]'
    >>> sys.path
    ['/home/mitchell/.../ko-rel/dist/python/lib/python27.zip', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/plat-linux2', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/lib-tk', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/lib-old', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/lib-dynload', 
     '/home/mitchell/.../ko-rel/dist/bin/python',
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/site-packages',
     '/home/mitchell/.../ko-rel/dist/bin/python/komodo',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/xml_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/rhtml_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/mxml_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/twig_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/django_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/xbl_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/php_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/mason_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/html_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/smarty_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/templatetoolkit_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/mustache_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/xslt_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/xul_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/refactoring@activestate.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/editorconfig@activestate.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/httpinspector@activestate.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/mercurial@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/git@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/bazaar@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/zendframework@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/golang@activestate.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/dbexplorer@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/dbexplorer_mysql@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/dbexplorer_mysql@ActiveState.com/platform/Linux_x86_64-gcc3/pylib27',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/dbexplorer_mysql@ActiveState.com/platform/Linux_x86_64-gcc3/pylib']

**Comment by mitchell-as** [comment]
## System Pyshell

    >>> import sys
    >>> sys.version
    '2.7.6 (default, Jun 22 2015, 17:58:13) \n[GCC 4.8.2]'
    >>> sys.executable
    '/usr/bin/python'
    >>> sys.path
    ['', 
     '/usr/lib/python2.7',
     '/usr/lib/python2.7/plat-x86_64-linux-gnu',
     '/usr/lib/python2.7/lib-tk',
     '/usr/lib/python2.7/lib-old',
     '/usr/lib/python2.7/lib-dynload',
     '/usr/local/lib/python2.7/dist-packages',
     '/usr/lib/python2.7/dist-packages',
     '/usr/lib/python2.7/dist-packages/PILcompat',
     '/usr/lib/python2.7/dist-packages/gtk-2.0',
     '/usr/lib/python2.7/dist-packages/ubuntu-sso-client']

## Komodo Pyshell

    >>> import sys
    >>> sys.executable
    '/usr/bin/python'
    >>> sys.version
    '2.7.8 (default, Nov 24 2014, 11:21:38) \n[GCC 4.8.2 20140120 (Red Hat 4.8.2-15)]'
    >>> sys.path
    ['/home/mitchell/.../ko-rel/dist/python/lib/python27.zip', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/plat-linux2', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/lib-tk', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/lib-old', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/lib-dynload', 
     '/home/mitchell/.../ko-rel/dist/bin/python',
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/site-packages',
     '/home/mitchell/.../ko-rel/dist/bin/python/komodo',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/xml_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/rhtml_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/mxml_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/twig_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/django_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/xbl_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/php_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/mason_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/html_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/smarty_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/templatetoolkit_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/mustache_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/xslt_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/xul_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/refactoring@activestate.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/editorconfig@activestate.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/httpinspector@activestate.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/mercurial@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/git@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/bazaar@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/zendframework@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/golang@activestate.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/dbexplorer@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/dbexplorer_mysql@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/dbexplorer_mysql@ActiveState.com/platform/Linux_x86_64-gcc3/pylib27',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/dbexplorer_mysql@ActiveState.com/platform/Linux_x86_64-gcc3/pylib']

**Comment by mitchell-as** [comment]
Sorry, I still don't buy it. Here's some information on my system Python vs. Komodo's Python. Please post the same output for your machine.

## System Pyshell

    >>> import sys
    >>> sys.version
    '2.7.6 (default, Jun 22 2015, 17:58:13) \n[GCC 4.8.2]'
    >>> sys.executable
    '/usr/bin/python'
    >>> sys.path
    ['', 
     '/usr/lib/python2.7',
     '/usr/lib/python2.7/plat-x86_64-linux-gnu',
     '/usr/lib/python2.7/lib-tk',
     '/usr/lib/python2.7/lib-old',
     '/usr/lib/python2.7/lib-dynload',
     '/usr/local/lib/python2.7/dist-packages',
     '/usr/lib/python2.7/dist-packages',
     '/usr/lib/python2.7/dist-packages/PILcompat',
     '/usr/lib/python2.7/dist-packages/gtk-2.0',
     '/usr/lib/python2.7/dist-packages/ubuntu-sso-client']

## Komodo Pyshell

    >>> import sys
    >>> sys.executable
    '/usr/bin/python'
    >>> sys.version
    '2.7.8 (default, Nov 24 2014, 11:21:38) \n[GCC 4.8.2 20140120 (Red Hat 4.8.2-15)]'
    >>> sys.path
    ['/home/mitchell/.../ko-rel/dist/python/lib/python27.zip', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/plat-linux2', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/lib-tk', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/lib-old', 
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/lib-dynload', 
     '/home/mitchell/.../ko-rel/dist/bin/python',
     '/home/mitchell/.../ko-rel/dist/python/lib/python2.7/site-packages',
     '/home/mitchell/.../ko-rel/dist/bin/python/komodo',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/xml_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/rhtml_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/mxml_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/twig_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/django_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/xbl_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/php_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/mason_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/html_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/smarty_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/templatetoolkit_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/mustache_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/xslt_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/distribution/bundles/xul_language@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/refactoring@activestate.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/editorconfig@activestate.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/httpinspector@activestate.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/mercurial@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/git@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/bazaar@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/zendframework@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/golang@activestate.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/dbexplorer@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/dbexplorer_mysql@ActiveState.com/pylib',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/dbexplorer_mysql@ActiveState.com/platform/Linux_x86_64-gcc3/pylib27',
     '/home/mitchell/.../ko-rel/dist/bin/extensions/dbexplorer_mysql@ActiveState.com/platform/Linux_x86_64-gcc3/pylib']

**Comment by Naatan** [comment]
I'll do you one better;

I renamed both python executables at `komodo/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/python/bin` and then ran Komodo, and well.. it ran.

**Comment by mitchell-as** [comment]
Please supply the information I requested. Your response doesn't help shed more light on the problem.

**Comment by Naatan** [comment]
```
sys.executable
'/usr/bin/python'
sys.version
'2.7.8 (default, Nov 24 2014, 11:21:38) \n[GCC 4.8.2 20140120 (Red Hat 4.8.2-15)]'
sys.path
['/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/python/lib/python27.zip', '/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/python/lib/python2.7', '/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/python/lib/python2.7/plat-linux2', '/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/python/lib/python2.7/lib-tk', '/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/python/lib/python2.7/lib-old', '/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/python/lib/python2.7/lib-dynload', '/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/python', '/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/python/lib/python2.7/site-packages', '/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/python/komodo', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/distribution/bundles/django_language@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/distribution/bundles/php_language@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/distribution/bundles/smarty_language@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/distribution/bundles/mxml_language@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/distribution/bundles/rhtml_language@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/distribution/bundles/xml_language@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/distribution/bundles/mason_language@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/distribution/bundles/xbl_language@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/distribution/bundles/xslt_language@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/distribution/bundles/html_language@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/distribution/bundles/xul_language@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/distribution/bundles/templatetoolkit_language@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/distribution/bundles/twig_language@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/distribution/bundles/mustache_language@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/extensions/editorconfig@activestate.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/extensions/dbexplorer@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/extensions/bazaar@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/extensions/golang@activestate.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/extensions/httpinspector@activestate.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/extensions/mercurial@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/extensions/git@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/extensions/refactoring@activestate.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/extensions/dbexplorer_mysql@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/extensions/zendframework@ActiveState.com/pylib', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/extensions/dbexplorer_mysql@ActiveState.com/platform/Linux_x86_64-gcc3/pylib27', u'/home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/bin/extensions/dbexplorer_mysql@ActiveState.com/platform/Linux_x86_64-gcc3/pylib']
```

Note the version of Python that is packaged with Komodo is 2.7.10, and so is my system version. No idea where this is getting 2.7.8 from ?

**Comment by Naatan** [comment]
Oh, found something.

This is python 2.7.8: `mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/python/bin/python` 
This is python 2.7.10: `mozilla/build/moz3500-ko9.10/mozilla/ko-rel/dist/python/bin/python2`

Neither are symlinked. When I launched komodo with a renamed python I had renamed both.

**Comment by mitchell-as** [comment]
Komodo appears to be using 2.7.8, as specified in `ko-rel/dist/python/include/python2.7/patchlevel.h`:

    #define PY_MAJOR_VERSION	2
    #define PY_MINOR_VERSION	7
    #define PY_MICRO_VERSION	8
    #define PY_VERSION      	"2.7.8"

**Comment by mitchell-as** [comment]
Okay, so you've confirmed that your Komodo is reporting version 2.7.8, which is the same as my Komodo. You've also stated your system Python is version 2.7.10, which obviously is not 2.7.8. Therefore it does not appear that your Komodo is using your system Python.

I know you want to say that `sys.executable` says otherwise, but I honestly don't know how trustworthy it is.

**Comment by Naatan** [comment]
Still, how do you explain that Komodo runs even though I renamed the packaged Python version?

**Comment by mitchell-as** [comment]
I went ahead and renamed my system Python executable and Komodo still runs. `sys.version` still returns 2.7.8. However, `sys.executable` now reports `''` (empty string).

**Comment by mitchell-as** [comment]
I am thinking that `sys.executable` is indeed unreliable and you renamed a file that didn't affect Komodo.

**Comment by Naatan** [comment]
That's my point; the file I renamed "should" affect Komodo. The fact that Komodo still runs implies it's using a different Python executable, and as there is no other Python executable with Komodo (except for the 2.7.3 one in the virtualenv folder), it has to be using something else?

**Comment by mitchell-as** [comment]
After internal discussion we have concluded that:

(1) The Komodo main process embeds its own Python interpreter and uses shared libraries instead of an executable. This explains why `sys.executable` is unreliable.
(2) The Python executable bundled with Komodo is used for out-of-process Python things like codeintel. Removing or renaming that executable will break codeintel.

The conclusion is that Komodo indeed is using its own version and not the system Python.

Closing this ticket.
